ci-templates/gitlab/filter-derive.yml

```diff
@@ -2,66 +2,73 @@
 # SPDX-License-Identifier: Apache-2.0
 # yaml-language-server: $schema=https://gitlab.com/gitlab-org/gitlab/-/raw/master/app/assets/javascripts/editor/schema/ci.json
 
-.derive:
-  rules:
-    - if: $CI_COMMIT_BRANCH
+derive:
+  image:
+    name: $DOCKER_REGISTRY/capella/cli:${CAPELLA_DOCKER_IMAGES_TAG}
+    entrypoint: [""]
 
-  script: &script-derive
-    - if [[ -n $DERIVE_RESULTS ]]; then mapfile -d ";" -t derive_results < <(echo -n "$DERIVE_RESULTS"); else derive_results=(); fi
+  script:
+    - |
+      if [[ -n $DERIVE_RESULTS ]]; then
+        mapfile -d ";" -t derive_results < <(echo -n "$DERIVE_RESULTS");
+      else
+        derive_results=();
+      fi
 
     # Reset branch. This is helpful when chaining jobs
     # and a previous job pushes changes that should be
     # considered for the model badge generation.
-    - git fetch
-    - git reset --hard origin/$CI_COMMIT_BRANCH
-
-    - pip install "capellambse[cli] @ git+https://github.com/DSD-DBS/py-capellambse.git@$CAPELLAMBSE_REVISION"
-
-    - python -m capellambse.extensions.filtering derive -m "${ENTRYPOINT:?}" -o derived-projects -p '{result.name}' "${derive_results[@]/#/--result=}" "$(if [[ -n $CAPELLA_DOCKERIMAGE ]]; then echo "--docker=$CAPELLA_DOCKERIMAGE"; else echo "--exe=$CAPELLA_EXECUTABLE"; fi)"
+    - |
+      if [[ -n $CI_COMMIT_BRANCH ]]; then
+        git fetch;
+        git reset --hard origin/$CI_COMMIT_BRANCH;
+      fi
+    - pip install "capellambse[cli] @ git+https://github.com/DSD-DBS/py-capellambse.git@${CAPELLAMBSE_REVISION:-release-0.5}"
+    - export DISPLAY=:99
+    - |
+      xvfb-run python \
+        -m capellambse.extensions.filtering derive \
+        -m "${ENTRYPOINT:?}" \
+        -o derived-projects \
+        -p '{result.name}' \
+        "${derive_results[@]/#/--result=}" \
+        --exe="/opt/capella/capella"
+    - |
+      git config --global user.email "${MODEL_MODIFIER_EMAIL:?}";
+      git config --global user.name "Filtering model modifier";
+      mkdir .git/info && echo "derived-projects/" > .git/info/exclude
+      if [[ "${PUSH_DERIVED_MODELS:-1}" == "1" ]]; then
+        for result in derived-projects/*; do
+          RESULT_DIR_NAME="${result#derived-projects/}"
+          DERIVED_BRANCH_NAME="$(echo "derived/$RESULT_DIR_NAME" | sed 's/[^a-zA-Z0-9.-_]/-/g')";
+          git switch "$DERIVED_BRANCH_NAME" || git switch --orphan "$DERIVED_BRANCH_NAME";
+          cp -r "$result/." .
+          git add .
+          git commit -m "feat: Add/update derived model '$RESULT_DIR_NAME'"
+          git push --set-upstream "https://${GIT_USERNAME:?}:${GIT_PASSWORD:?}@${CI_SERVER_HOST}/${CI_PROJECT_PATH}.git" "$DERIVED_BRANCH_NAME"
```

Comment:
> Since we're pushing back into the same repo, we should be able to simply say `git push origin "$DERIVED_BRANCH_NAME"`, which uses the `$CI_JOB_TOKEN` for authentication – did you try that out? This would allow us to get rid of the `GIT_PASSWORD` environment variable, at least in this job template.

This doesn't work. It's an open issue and even though the documentation states that is *should* work (["The token has the same permissions to access the API as the user that caused the job to run."](https://docs.gitlab.com/ee/ci/jobs/ci_job_token.html)), it doesn't: https://gitlab.com/gitlab-org/gitlab/-/issues/389060

> Also, don't use `--set-upstream` if you're passing passwords around, it'll end up unencrypted in the git config file.
Makes sense.

Reply:
The model badge template also sets the upstream. I'll remove it there too.